source/DurationPicker.js

```diff
@@ -0,0 +1,516 @@
+		},
+
+		events: {
+			/**
+			* {@link moon.DurationPicker#event:onDurationChange}
+			*/
+			onDurationChange: '',
+
+			/**
+			* {@link moon.DurationPicker#event:onCountdownExpired}
+			*/
+			onCountdownExpired: ''
+		},
+
+		/**
+		* @public
+		* @lends moon.DurationPicker.prototype
+		*/
+		published: {
+
+			/**
+			* Text to be displayed as the current value if no item is currently selected
+			*
+			* @type {String}
+			* @default ''
+			* @public
+			*/
+			noneText: '',
+
+			/**
+			* The value of the picker, expressed as a standard JavaScript 
+			* object.
+			*
+			* @type {String}
+			* @default null
+			* @public
+			*/
+			value: null,
+
+			/**
+			* Optional label for hour
+			*
+			* @type {String}
+			* @default moon.$L('hour')
+			* @public
+			*/
+			hourText: moon.$L('hour'),			// i18n 'HOUR' label in moon.DurationPicker widget
+
+			/**
+			* Optional label for minute
+			*
+			* @type {String}
+			* @default moon.$L('minute')
+			* @public
+			*/
+			minuteText: moon.$L('minute'),		// i18n 'MINUTE' label in moon.DurationPicker widget
+
+			/**
+			* Optional label for second
+			*
+			* @type {String}
+			* @default moon.$L('second')
+			* @public
+			*/
+			secondText: moon.$L('second'),		// i18n 'SECOND' label in moon.DurationPicker widget
+
+			/**
+			* @type {Boolean}
+			* @default ''
+			* @public
+			*/
+			countdown: false,
+
+			/**
+			* @type {String}
+			* @default ''
+			* @public
+			*/
+			template: ''
+		},
+
+		/**
+		* @type {String}
+		* @default null
+		* @private
+		*/
+		timer: null,
+
+		/**
+		* @type {Object}
+		* @default '0, 0, 0'
+		* @private
+		*/
+
+		valueArray: {},
+
+		/**
+		* @private
+		*/
+		components: [
+			{name: 'headerWrapper', kind: 'moon.Item', classes: 'moon-date-picker-header-wrapper', onSpotlightFocus: 'headerFocus', ontap: 'expandContract', components: [
+				// headerContainer required to avoid bad scrollWidth returned in RTL for certain text widths (webkit bug)
+				{name: 'headerContainer', classes: 'moon-expandable-list-item-header moon-expandable-picker-header moon-expandable-datetime-header', components: [
+					{name: 'header', kind: 'moon.MarqueeText'}
+				]},
+				{name: 'currentValue', kind: 'moon.MarqueeText', classes: 'moon-expandable-picker-current-value'}
+			]},
+			{name: 'drawer', kind: 'enyo.Drawer', resizeContainer: false, classes: 'moon-expandable-list-item-client indented', components: [
+				{name: 'client', kind: 'enyo.Control', classes: 'enyo-tool-decorator moon-date-picker-client', onSpotlightLeft: 'closePicker', onSpotlightSelect: 'closePicker'}
+			]}
+		],
+
+		/**
+		* @private
+		*/
+		bindings: [
+			{from: '.disabled', to: '.$.headerWrapper.disabled'},
+			{from: '.disabledPicker', to: '.$.hour.disabled'},
+			{from: '.disabledPicker', to: '.$.minute.disabled'},
+			{from: '.disabledPicker', to: '.$.second.disabled'}
+		],
+
+		/**
+		* @private
+		*/
+		constructor: function() {
+			this.inherited(arguments);
```

Comment:
While `enyo.inherit` is technically faster, `this.inherited` makes the code easier to read. We could, during minification for example, automatically upgrade each `this.inherited` to a faster `enyo.inherit` method, because it's so consistent and easily readable. There won't be hundreds of these stamped out at a time on a single view, so the speed benefit is minimal. Either of these options is technically fine though.